src/objective-c/tests/CoreTests/CoreTests/EventEngineTimerTests.mm

```diff
@@ -40,12 +47,87 @@ + (void)tearDown {
 }
 
 - (void)setUp {
+  signaled_ = false;
+  _engine = absl::make_unique<grpc_event_engine::experimental::LibuvEventEngine>();
 }
 
 - (void)tearDown {
+  _engine = nullptr;
+}
+
+- (void)testImmediateCallbackIsExecutedQuickly {
+  grpc_core::MutexLock lock(&mu_);
+  _engine->RunAt(absl::Now(), [self]() {
+    grpc_core::MutexLock lock(&mu_);
+    signaled_ = true;
+    cv_.Signal();
+  });
+  cv_.WaitWithTimeout(&mu_, absl::Seconds(5));
```

Comment:
Just curious -- is 5 seconds considered "executed quickly"? :)

Reply:
It could probably be lowered safely. I think of this more as a failsafe than an expectation. Setting it too low, we risk making this a flaky test if our test infra is feeling under the weather any given day.

Reply:
Yeah makes sense.. And FYI I only made this comment coz the test method is called "...ExecutedQuickly" :D